HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+     * This function takes an array of roll, pitch and yaw angles, and a convention, as input parameters.
+     * For rollPitchYaw we define that \a roll is a rotation about x-axis,
+     * \a pitch is a rotation about y-axis and \a yaw is a rotation about z-axis.
+     * @param rollPitchYaw Roll, Pitch and Yaw angles in radians. Order: Ox, Oy, Oz
+     * @param rotation Enum to indicate which convention to follow for this conversion
+     * @return rotationMatrix
+     */
+    Eigen::Matrix3d rpy2rotmat(const Eigen::Array3d rollPitchYaw,
+                               const TransformUtils::Rotation rotation = ZYX_INTRINSIC)
+    {
+        Eigen::Matrix3d rotationMatrix = Eigen::Matrix3d::Identity(3, 3);
+
+        // rollPitchYaw is ordered by Ox, Oy and Oz. Where Oa is the rotation about axis a.
+        switch(rotation)
+        {
+            case XYZ_INTRINSIC:
+            case ZYX_EXTRINSIC:
+                rotationMatrix = Eigen::AngleAxisd(rollPitchYaw[0], Eigen::Vector3d::UnitX())
+                                 * Eigen::AngleAxisd(rollPitchYaw[1], Eigen::Vector3d::UnitY())
+                                 * Eigen::AngleAxisd(rollPitchYaw[2], Eigen::Vector3d::UnitZ());
+                break;
+            case ZYX_INTRINSIC:
+            case XYZ_EXTRINSIC:
+                rotationMatrix = Eigen::AngleAxisd(rollPitchYaw[2], Eigen::Vector3d::UnitZ())
+                                 * Eigen::AngleAxisd(rollPitchYaw[1], Eigen::Vector3d::UnitY())
+                                 * Eigen::AngleAxisd(rollPitchYaw[0], Eigen::Vector3d::UnitX());
+                break;
+            default: throw "Invalid orientation";
+        }
+
+        return rotationMatrix;
+    }
+
+    int compareRotation(const Eigen::Matrix3d rotationMatrix1, const Eigen::Matrix3d rotationMatrix2)
+    {
+        std::cout << "\nAbsolute error introduced by transformation" << std::endl;
+        const auto rotationMatrixError = (rotationMatrix1 - rotationMatrix2).cwiseAbs();
+        std::cout << rotationMatrixError << std::endl;
+
+        std::cout << "\nRelative (elementwise percent) error introduced by transformation" << std::endl;
+        std::cout << (rotationMatrixError.cwiseQuotient(rotationMatrix1)) * 100 << std::endl;
+
+        std::cout << "\nRotation difference introduced by transformation" << std::endl;
+        const auto differenceRotationMatrix = rotationMatrix1 * rotationMatrix2.inverse();
+        std::cout << differenceRotationMatrix << std::endl;
+
+        std::cout << "\nRotation distance introduced by transformation" << std::endl;
+        const auto axisAngle = Eigen::AngleAxisd(differenceRotationMatrix);
+        auto rotationDistance = axisAngle.angle();
+        std::cout << rotationDistance << std::endl;
+
+        const auto quaternion1 = Eigen::Quaterniond(rotationMatrix1);
+        const auto quaternion2 = Eigen::Quaterniond(rotationMatrix2);
+        rotationDistance = 2 * acos((quaternion1 * quaternion2.conjugate()).w());
```

Comment:
I think we don't need two ways to calculate rotationDistance. They should give the same results. Choose one, delete the other?